core/src/smartcontracts/isi/triggers.rs

```diff
@@ -81,6 +91,9 @@ pub mod isi {
             wsv: &WorldStateView,
         ) -> Result<(), Self::Error> {
             let trigger_id = self.object_id.clone();
+
+            wsv.world.domain_scoped_triggers.remove(&trigger_id);
```

Comment:
This is an important point. regardless of domain, triggers have unique IDs. So no matter whether a trigger is global, or specific to a domain, no two triggers can have the same ID. So my question is, do we consider the domain part of the ID?

Reply:
Yes, a trigger id is unique in one domain scope.

So your suggest is product two types `TriggerId = Id x Option<DomainId>`, and the TriggerId structure will look like this globally, right?
```
pub struct Id {
    /// Name given to trigger by its creator.
    pub name: Name,
    /// Refer to already exists domain
    pub domain_id: Option<DomainId>,
}
```

Reply:
@appetrosyan please reply to this comment.

Reply:
Yes, effectively. The option is not necessary in general, it's just useful until the point where we put the trigger into the `Set`.